engine/lib/phx/script/ffiext/Vec2.lua

```diff
@@ -153,14 +153,18 @@ local function defineVec2_t(t, mt)
 
     function mt.__index.inormalize(v)
         local l = sqrt(v.x * v.x + v.y * v.y)
-        assert(l > 1e-12, 'Attempting to normalize vector with near-zero length')
+        if l < 1e-12 then
+            l = 0
```

Comment:
this will end up dividing by zero and creating a NaN vector. is that intended?

Reply:
you are right, does this sound more reasonable to you?

```lua
    function mt.__index.inormalize(v)
        local l = sqrt(v.x * v.x + v.y * v.y)
        if l < 1e-12 then
            v.x = 0
            v.y = 0
        else
            v.x = v.x / l
            v.y = v.y / l
        end
    end

    function mt.__index.normalize(v)
        local l = sqrt(v.x * v.x + v.y * v.y)
        if l < 1e-12 then
            return Vec2(0, 0)
        end
        return (Vec2(v.x / l, v.y / l))
    end
    ```

Reply:
Yup looks better, but why are we supporting 0 length vectors anyway? Seems like something is wrong somewhere else right?

Reply:
You might need to apply the same change to Vec3/Vec4 too

Reply:
> Yup looks better, but why are we supporting 0 length vectors anyway? Seems like something is wrong somewhere else right?

Probably. It happens quite often currently with a lot of ships.

Reply:
usually in this line https://github.com/Limit-Theory-Redux/ltheory/blob/0c6e9ad7c3b84c676a999f10b6566ba2b21c8395/script/GameObjects/Actions/Attack.lua#L84

Reply:
In that case I suspect in some cases `self.target == e` (I guess the enemy we're attacking _is_ the target? dont have context on the code). Maybe it's better if we add a condition here to do something different if self.target == e? I don't particularly like the idea of adding a branch to `normalize`, as it's hitting a pretty hot code path

Reply:
will do that, reverting all changes to the ffiext files too